holonote/app/panel.py

```diff
@@ -224,57 +224,69 @@ def _callback_apply(self, event):
+        """
+        Prevents multiple layouts from being visible at the same time.
+        """
+        desired_layout = None
+        with param.parameterized.batch_call_watchers(self):
+            for name, layout in self._layouts.items():
+                if name == desired_name:
+                    layout.visible = True
+                    desired_layout = layout
+                elif not name.startswith("panel"):
+                    layout.visible = False
+
+        # If the desired layout is not found, create it
+        if desired_name is not None and desired_layout is None:
+            desired_layout = self._add_layout(desired_name)
+        return desired_layout
 
     def _register_stream_popup(self, stream):
         def _popup(*args, **kwargs):
-            layout = self._get_layout(stream.name)
-            with param.parameterized.batch_call_watchers(self):
-                self._hide_layouts()
-                self._widget_mode_group.value = "+"
-                layout.visible = True
-                return layout
+            # If the annotation widgets are laid out on the side,
+            # do not show the popup when in subtract or edit mode
+            widgets_on_side = any(name.startswith("panel") for name in self._layouts)
+            if widgets_on_side and self._widget_mode_group.value in ("-", "✏"):
+                return
+            self._widget_mode_group.value = "+"
+            return self._hide_layouts_except(stream.name)
 
         stream.popup = _popup
 
     def _register_tap_popup(self, display):
         def tap_popup(x, y) -> None:  # Tap tool must be enabled on the element
-            layout = self._get_layout("tap")
-            if self.annotator.selection_enabled:
-                with param.parameterized.batch_call_watchers(self):
-                    self._hide_layouts()
-                    layout.visible = True
-                    return layout
+            if self.annotator.selected_indices:
+                return self._hide_layouts_except("tap")
 
         display._tap_stream.popup = tap_popup
 
     def _register_double_tap_clear(self, display):
         def double_tap_toggle(x, y):
-            layout = self._get_layout("doubletap")
-            if layout.visible:
-                with param.parameterized.batch_call_watchers(self):
-                    self._hide_layouts()
-                    layout.visible = True
-                    return layout
+            # Toggle the visibility of the doubletap layout
+            if any(layout.visible for layout in self._layouts.values()):
+                # Clear all open layouts
+                self._hide_layouts_except(None)
+            else:
+                # Open specifically the doubletap layout
+                return self._hide_layouts_except("doubletap")
 
-        try:
-            tools = display._element.opts["tools"]
-        except KeyError:
-            tools = []
-        display._element.opts(tools=[*tools, "doubletap"])
```

Comment:
Did this not work? I am heavily in favor of keeping this.

Reply:
Just seemed like a lot of logic. I can move it back.

Reply:
But you are adding functionality that is not needed in the annotator class; this is purely a tool the widgets need to function.

Please don't make unnecessary changes without mentioning the changes have nothing to do with the issue you are trying to fix in this PR.   This can be done as a comment.

Reply:
At first it was not functioning properly which ish why I tried that. Anyhow, I've reverted it already.